lib/streams/test.js: convert hex strings to ascii

dbrock opened the pull request:
This change turns the following test output:

```
MakerTest
  test basic locking and freeing
  LOG:  log_bytes32
  LOG:    val: 0x4e6f7420657175616c2100000000000000000000000000000000000000000000
  LOG:  0x4100000000000000000000000000000000000000000000000000000000000000: 100
  LOG:  0x4200000000000000000000000000000000000000000000000000000000000000: 70
  LOG:  log_bytes32
  LOG:    val: 0x4e6f7420657175616c2100000000000000000000000000000000000000000000
  LOG:  0x4100000000000000000000000000000000000000000000000000000000000000: 100
  LOG:  0x4200000000000000000000000000000000000000000000000000000000000000: 80
  Failed!
```

into this:

```
MakerTest
  test basic locking and freeing
  LOG:  log_bytes32
  LOG:    val: Not equal!
  LOG:  A: 100
  LOG:  B: 70
  LOG:  log_bytes32
  LOG:    val: Not equal!
  LOG:  A: 100
  LOG:  B: 80
  Failed!
```

See any problems with this @ryepdx @denis @nikolai?

mhhf commented:
how did i react with a confused smily when my github token is mhhf ?!
My world just shifted left...
Oh, you pinged the wrong person, and he responded!
haha
i like this fix 👍

dbrock commented:
Ahh sorry lol.

dbrock commented:
I reverted the change to the else clause and introduced a new clause instead in order to treat `log_bytes32` specially.

Here's what the output looks like now:

```
MakerTest
  test basic locking and freeing
  | Error: Wrong `uint' value
  |   Actual: 100
  |   Expected: 70
  | Error: Wrong `uint' value
  |   Actual: 100
  |   Expected: 80
  Failed!
```

dbrock commented:
Added a new function `assertEqDecimal(uint actual, uint expected, uint decimals)` in order to be able to format monetary values nicely, like this:

```
MakerTest
  test issuing and covering
  | Error: Wrong fixed-point decimal
  |   Expected: 28.500000000000000000
  |   Actual:   11248948841052480191659961171643302847826033.276968093805044432
  | Error: Wrong fixed-point decimal
  |   Expected: 2500.000000000000000000
  |   Actual:   11248948841052480191659961171643302847826033.276968093805044432
  Failed!
```

dbrock commented:
If nobody objects I think I'll go ahead and merge this soon.

mhhf commented:
:+1: for merge

dbrock commented:
@mhhf Cool, thanks for taking a look!

rainbeam commented:
This is great! My eyes are happy now.

nikolai commented:
@dbrock you've probably mentioned me by mistake...

dbrock commented:
@nikolai yeah, sorry, I'm an idiot! great username btw

nikolai commented:
never mind, glad to meet your project :)
good luck
